### PR TITLE
[#19] Login redirect state url fixed

### PR DESCRIPTION
**Pull request by nirojdyola**

### CODE REVIEW

File: src/app/login/login.component.ts [hunk 1]
```diff
@@ -32,30 +32,36 @@ require('./login.scss');
...
+export class LoginComponent {
     constructor(
         private couchService: CouchService,
-        private router: Router
+        private router: Router,
+        private route: ActivatedRoute
     ) {}
-    
+
+    ngOnInit() {
```
Comment (paulbert) [inline src/app/login/login.component.ts]: I don't think you actually should use ngOnInit here, just assign returnUrl to the route.

ngOnInit is used mainly for accessing the services to initialize data.  `returnUrl` can be set with just simple assignment, so it doesn't need to utilize that lifecycle hook.

Comment (nirojdyola) [reply src/app/login/login.component.ts]: Used returnUrl in constructor

**Comment by paulbert** [comment]
You don't need to add it into the Constructor, either.  Just add it as a normal property of the class.

**Comment by paulbert** [comment]
@nirojdyola See my latest commit.  `returnUrl` can be set immediately.  You don't need to set it in a method.

**Pull request merged**